### PR TITLE
added header, banner & navigation

### PR DESCRIPTION
**Pull request by Nilegfx**
- [x] added header, banner & navigation

https://m3-fitness-zahraa-khalil.web.app/

### CODE REVIEW

File: dist/assests/css/header-banner-nav.css [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+.main-header__nav ul li {
+  display: inline-block;
+  padding: 0.8rem 1rem;
+}
+.main-header__nav ul,
+.main-header__nav ul li a {
+  text-decoration: none;
+  color: white;
+}
+.logo a {
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css:37]: in the original mockups (the following pic) 
Original Mockups
image

This PR implementation
image


the logos is vertically centered with the left navigation, it would be nice if you can fix the alignement

File: dist/assests/css/header-banner-nav.css [hunk 2]
```diff
@@ -0,0 +1,161 @@
+@import url('header-banner-nav.css');
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: Why?

File: dist/assests/css/header-banner-nav.css [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+/* Main Section, Header, Nav */
+.main-header {
+  background-image: url('../img/banner.jpg');
+  height: 100vh;
+  background-repeat: no-repeat;
+  background-size: cover;
+  position: relative;
+  width: 100%;
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: width is not needed here.

File: dist/assests/css/header-banner-nav.css [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+}
+
+.main-header .main-header__nav {
+  /* background-color: blue; */
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: please remove this comment

File: dist/assests/css/header-banner-nav.css [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-items: baseline;
+  width: 100%;
+  margin-right: 1rem;
+  font-size: 11px;
+  font-weight: 600;
+  letter-spacing: 0.02em;
+  position: absolute;
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: this line and the three following properties are not needed.

File: dist/assests/css/header-banner-nav.css [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  top: 0;
+  left: 0;
+  z-index: 40;
+}
+.main-header__nav .left {
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: `.left` is not descriptive, better to name it something along the lines `main-navigation` or `left-navigation`

File: dist/assests/css/header-banner-nav.css [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  margin-left: 2rem;
+}
+.main-header__nav .right {
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: [same comment as this one ](https://github.com/m3ntorship/fitness/pull/11/files#r407259758)

File: dist/assests/css/header-banner-nav.css [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+.logo a {
+  text-decoration: none;
+  color: white;
+  font-size: 20px;
+  font-weight: 700;
+}
+.logo span {
+  color: #fb7c2f;
+}
+.btn2 {
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: [same comment as this one ](https://github.com/m3ntorship/fitness/pull/11/files#r407259758), in this case, the more descriptive name should be `btn-secondary` giving that we already have `btn-primary`
using the native cascading nature of css, you can start styling this button from the primary "orange" button, remove the background and give it a white border.

File: dist/assests/css/header-banner-nav.css [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+  height: auto;
+  text-align: center;
+  font-size: smaller;
+  width: auto;
+  border: 1px solid white;
+  letter-spacing: 0.2em;
+}
+.main-section__info {
+  position: absolute;
+  z-index: 30;
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css:59]: why?

File: dist/assests/css/header-banner-nav.css [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+  top: 24%;
+  left: 15%;
+  overflow: hidden;
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: why?

File: dist/assests/css/header-banner-nav.css [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+  width: auto;
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: why?

File: dist/assests/css/header-banner-nav.css [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+}
+.show-bar{
+  display: none;
+}
+.main-section__info p {
+  margin-top: -1.7rem;
+  padding-bottom: 3.5rem;
+  letter-spacing: 0.03em;
+  font-size: smaller;
+  color: rgba(240, 248, 255, 0.883);
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: 0.8 will do the trick 😉

File: dist/assests/css/header-banner-nav.css [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+}
+#orange {
```
Comment (Nilegfx) [inline dist/assests/css/header-banner-nav.css]: e2fesh ID malhosh lazma 😂
Please don't use ids for stylistic purposes. always use classes.

File: dist/assests/css/header.css [hunk 14]
```diff
@@ -1,15 +1,190 @@
 /* Main Section, Header, Nav */
```
Comment (Nilegfx) [inline dist/assests/css/header.css:1]: this file is not used, please remove it.

File: dist/assests/css/header.css [hunk 15]
```diff
@@ -1,15 +1,190 @@
...
-.main-header .main-header__nav {
+
```
Comment (Nilegfx) [inline dist/assests/css/header.css]: this file is not used, please remove it.

File: dist/index.html [hunk 16]
```diff
@@ -3,14 +3,20 @@
   <head>
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Fitness</title>
```
Comment (Nilegfx) [inline dist/index.html:6]: Thanks for adding the missing title.

File: dist/index.html [hunk 17]
```diff
@@ -3,14 +3,20 @@
...
     <link
+      href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@300;600;700;800&display=swap"
```
Comment (Nilegfx) [inline dist/index.html:8]: Why did you remove integrity and crossorigin attributes?

File: dist/index.html [hunk 18]
```diff
@@ -3,14 +3,20 @@
...
       rel="stylesheet"
-      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.13.0/css/all.min.css"
-      integrity="sha256-h20CPZ0QyXlBuAw7A+KluUYx/3pK+c7lYEpqLTlxjYQ="
-      crossorigin="anonymous"
+    />
+    <link
+      rel="stylesheet"
+      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css"
+    />
+    <link
```
Comment (Nilegfx) [inline dist/index.html]: I think this is a duplicate file, please remove it otherwise the page will request this file twice each time someone visits us.

File: dist/index.html [hunk 19]
```diff
@@ -20,30 +26,34 @@
...
-              FITNESS
+              <a href="#">FITNE<span class="orange">SS</span></a>
             </span>
-            <ul class="nav-links">
-              <li>link</li>
-              <li>link</li>
-              <li>link</li>
-              <li>link</li>
-              <li>link</li>
+            <ul class="nav-links hide">
```
Comment (Nilegfx) [inline dist/index.html]: the `hide` class is redundant here, doesn't have corresponding styles

File: dist/index.html [hunk 20]
```diff
@@ -20,30 +26,34 @@
...
+              <li><a href="#">HOME</a></li>
+              <li><a href="#">ABOUT US</a></li>
+              <li><a href="#">PAGE</a></li>
+              <li><a href="#">BLOG</a></li>
+              <li><a href="#">CONTACTS</a></li>
             </ul>
           </div>
-          <div class="nav__login right">
+          <div class="nav__login right hide">
```
Comment (Nilegfx) [inline dist/index.html]: [same comment as this one](https://github.com/m3ntorship/fitness/pull/11/files#r407261368)

File: dist/index.html [hunk 21]
```diff
@@ -20,30 +26,34 @@
...
+          </div>
         </nav>
+           
       </header>
 
       <div class="main-section__info">
-        <h1>Make Your Body Beautiful</h1>
-        <p>Lorem ipsum dolor sit amet.</p>
-        <button class="btn btn-primary">Book Now</button>
+        <h1>MAKE YOUR BODY <br /><span id="orange">BEAUTIFUL</span></h1>
```
Comment (Nilegfx) [inline dist/index.html]: [same as this comment](https://github.com/m3ntorship/fitness/pull/11/files#r407260592)

File: dist/assests/css/header-banner-nav.css (hunk 1 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css:37]: okk working on it

File: dist/index.html (hunk 17 above)
Comment (zahraa-khalil) [reply dist/index.html:8]: by mistake

File: dist/index.html (hunk 18 above)
Comment (zahraa-khalil) [reply dist/index.html]: No, one is .css and the other is .js both are required...

Comment (Nilegfx) [reply dist/index.html]: Then two things: 
- `<link>` tag doesn't work with `.js` files
- why do we need `js` for font-awesome at all?

**Comment by Nilegfx** [comment]
# it is important to delete the `header.css` and its import statement from the `styles.css`  
it makes this PR introduces a problem with all other components which are working just fine on live now.

# with the `header-banner-navigation` changes
image

# without the `header-banner-navigation` changes
image

File: dist/assests/css/header-banner-nav.css (hunk 1 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css:37]: Done

File: dist/assests/css/header-banner-nav.css (hunk 7 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css]: updated

File: dist/assests/css/header-banner-nav.css (hunk 9 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css:59]: to fix positioning for the main header

File: dist/assests/css/header-banner-nav.css (hunk 10 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css]: to fix positioning for the main header

File: dist/assests/css/header-banner-nav.css (hunk 12 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css]: Done :D

File: dist/assests/css/header-banner-nav.css (hunk 13 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css]: okkk :D

File: dist/assests/css/header.css (hunk 15 above)
Comment (zahraa-khalil) [reply dist/assests/css/header.css]: updated

File: dist/index.html (hunk 18 above)
Comment (zahraa-khalil) [reply dist/index.html]: removed

File: dist/index.html (hunk 19 above)
Comment (zahraa-khalil) [reply dist/index.html]: removed

**Review by zahraa-khalil — CHANGES_REQUESTED** [review]
Ready for review

File: dist/assests/css/header-banner-nav.css (hunk 11 above)
Comment (zahraa-khalil) [reply dist/assests/css/header-banner-nav.css]: deleted width & overflow
others needed for positioning...

**Review by zahraa-khalil — COMMENTED** [review]
review is required

**Review by zahraa-khalil — APPROVED** [review]

**Comment by zahraa-khalil** [comment]
127 0 0 1_5500_dist_index html (3)

**Review by zahraa-khalil — COMMENTED** [review]
done
